src/Http/Middleware/Google2fa.php

```diff
@@ -23,19 +32,46 @@ class Google2fa
      */
     public function handle($request, Closure $next)
     {
-        if (!config('lifeonscreen2fa.enabled')) {
+        if ($request->route()->getName() == 'nova.logout') {
+            return $next($request);
+        }
+
+        $response = $this->doHandle($request, $next);
+
+        return $this->preventBrowserCaching($response);
+    }
+
+    /**
+     * @param Request $request
+     * @param Closure $next
+     * @return mixed
+     */
+    private function doHandle(Request $request, Closure $next)
+    {
+        if ($emailDomain = config('lifeonscreen2fa.user_email_domain')) {
+            if (!strpos($request->user()->email, '@' . $emailDomain)) {
+                return $next($request);
+            }
+        }
+
+        if ( ! config('lifeonscreen2fa.enabled') || (config('lifeonscreen2fa.optional') && ($request->user()->user2fa === null || auth()->user()->user2fa->google2fa_enable === 0))) {
             return $next($request);
         }
-        if ($request->path() === 'los/2fa/confirm' || $request->path() === 'los/2fa/authenticate'
-            || $request->path() === 'los/2fa/register') {
+        if (
+            $request->path() === 'los/2fa/confirm'
+            || $request->path() === 'los/2fa/authenticate'
+            || $request->path() === 'los/2fa/register'
+            || $request->path() === 'los/2fa/recovery'
+        ) {
             return $next($request);
         }
         $authenticator = app(Google2FAAuthenticator::class)->boot($request);
         if (auth()->guest() || $authenticator->isAuthenticated()) {
             return $next($request);
         }
         if (empty(auth()->user()->user2fa) || auth()->user()->user2fa->google2fa_enable === 0) {
-
+//            $google2FaManager = app(Google2faManager::class);
+//            return $google2FaManager->showRecoveryView();
```

Comment:
Can you remove this commented out code please @NickSnellockNS

Reply:
All beyond that :+1:

Reply:
all gone